Sources/Plasma/CoreLib/plFileSystem.cpp

```diff
@@ -460,7 +469,36 @@ plFileName plFileSystem::GetUserDataPath()
 
         _userData = plFileName::Join(ST::string::from_wchar(path), plProduct::LongName());
 #else
-        _userData = plFileName::Join(getenv("HOME"), "." + plProduct::LongName());
+        ST::string dir;
+#  if HS_BUILD_FOR_APPLE
+        char home[256] = "";
+        if (__builtin_available(macOS 10.12, *)) {
+            sysdir_search_path_enumeration_state state;
+            state = sysdir_start_search_path_enumeration(SYSDIR_DIRECTORY_APPLICATION_SUPPORT, SYSDIR_DOMAIN_MASK_USER);
+            state = sysdir_get_next_search_path_enumeration(state, home);
+        } else {
+            #pragma clang diagnostic push
+            #pragma clang diagnostic ignored "-Wdeprecated-declarations"
+            NSSearchPathEnumerationState state;
+            state = NSStartSearchPathEnumeration(NSApplicationSupportDirectory, NSUserDomainMask);
+            state = NSGetNextSearchPathEnumeration(state, home);
+            #pragma clang diagnostic pop
+        }
+        dir = home;
+#  else
+        dir = "~" PATH_SEPARATOR_STR ".config";
+#  endif
+
+        wordexp_t p;
+        wordexp(dir.c_str(), &p, 0);
+        dir = p.we_wordv[0];
+        for (int i = 1; i < p.we_wordc; i++) {
+            dir += " ";
+            dir += p.we_wordv[i];
+        }
+        wordfree(&p);
```

Comment:
Looking at the man page, `wordexp` does a lot more than just expanding tildes to the home path - it also expands variables and globs, does shell-style word splitting, and might even try to interpret some other shell syntax. While realistically nobody will have shell metacharacters or weird whitespace in their home path, do we really need to use a complex function like this here?

I don't think we need to worry about the case of `HOME` being unset. The [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html#variables) implicitly assumes that `HOME` is always set:

> If `$XDG_CONFIG_HOME` is either not set or empty, a default equal to `$HOME/.config` should be used.

And [according to POSIX](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_01), expanding a plain tilde when `HOME` is unset has unspecified results:

> If the login name is null (that is, the tilde-prefix contains only the tilde), the tilde-prefix is replaced by the value of the variable *HOME*. If *HOME* is unset, the results are unspecified.

Reply:
For anyone else reading this and wondering why `wordexp` is also being called on the paths returned by the macOS-specific API - according to the man page:

> Directory paths returned in the user domain will contain "~" to refer to the user’s directory.

So if we don't use `wordexp`, we'd have to handle this case manually - but that should be relatively simple, because the tilde that needs to be expanded should only appear as the first character of the path.